### PR TITLE
Select parent skeleton first before other properties are suggested

### PR DESCRIPTION
**Pull request by p-spacek**
### What does this PR do?
This functionality helps a less technical person not to get lost on completion when there are complicated schemas and many properties that can be used (see the last screen - there are too many options so the user should select the parent/skeleton first and don't have to care about all possible properties).

It little bit extends 'parent/skeleton' object completion.

This PR adds`yaml.selectParentSkeletonFirst`. If true, the user must select some parent skeleton first before autocompletion starts to suggest the rest of the properties.
When yaml object is not empty, autocompletion ignores this setting and returns all properties and skeletons.

Not sure if it's useful for your users but it can help others (less technical creators).

### examples

suggest only skeletons because object is empty:
image

whit disabled property `yaml.selectParentSkeletonFirst` (default)
image

properties are suggested as usual when the user tries to write it:
image

when the object is not empty, standard completion is displayed
image

Note that I haven't created PR for yaml client yet. I can create it when you check this one.

### CODE REVIEW

**Comment by gorkem** [comment]
LGTM in general one minor comment on the code. However I would like to change the name of the setting. As we seem to be getting more settings related to completions. Can we start to group such settings under `suggest` ? For this setting it would be something like  `yaml.suggest.parentSkeletonSelectedFirst`.

**Comment by gorkem** [comment]
Also this one needs a rebase

File: src/languageservice/services/yamlCompletion.ts [hunk 1]
```diff
@@ -589,7 +591,8 @@ export class YamlCompletion {
     const lineContent = textBuffer.getLineContent(overwriteRange.start.line);
     const hasOnlyWhitespace = lineContent.trim().length === 0;
     const hasColon = lineContent.indexOf(':') !== -1;
-
+    const isNodeNull =
```
[the remark this answers is not in the document]
Comment (p-spacek) [reply src/languageservice/services/yamlCompletion.ts]: I moved it down neer to usage

**Comment by p-spacek** [comment]
> LGTM in general one minor comment on the code. However I would like to change the name of the setting. As we seem to be getting more settings related to completions. Can we start to group such settings under `suggest` ? For this setting it would be something like `yaml.suggest.parentSkeletonSelectedFirst`.

I moved `parentSkeletonSelectedFirst` into the new `suggest` object

**Review by gorkem — APPROVED** [review]

**Review by msivasubramaniaan — APPROVED** [review]

**Pull request merged**